ride/otc.ride

```diff
@@ -64,13 +75,29 @@ func currentWithdrawals(address: String, amount: Int) = {
     [toWithdrawWithoutFee, toWithdrawWithFee, fee]
 }
 
+func mustManager(i: Invocation) = {
+  let pd = "Permission denied".throw()
+
+  match managerPublicKeyOrUnit() {
+    case pk: ByteVector => i.callerPublicKey == pk || pd
+    case _: Unit => i.caller == this || pd
+  }
+}
+
+@Callable(i)
+func constructor(usdtAssetId: String, usdnAssetId: String) = {
+  [
+    StringEntry(keyUsdtAssetId(), usdtAssetId),
+    StringEntry(keyUsdnAssetId(), usdnAssetId)
+  ]
+}
 
 @Callable(invoke)
 func swapUSDTtoUSDN() =   {
     let balanceKey = collectBalanceKeyWithFee(invoke.caller.toString())
     let payment = value(invoke.payments[0])
 
-    if (payment.assetId != idUSDT) then usdtToUsdnAssetError() else
+    if (payment.assetId != keyUsdtAssetId().fromBase58String()) then usdtToUsdnAssetError() else
```

Comment:
You should read string value from state in proper way. This code makes `ByteVector` from string `%s__USDT`. Here and later in code same mistake
https://docs.waves.tech/en/ride/functions/built-in-functions/account-data-storage-functions